run.sh

```diff
@@ -120,6 +117,115 @@ function parse_ports() {
     done
 }
 
+function deploy_ear() {
+    
+    ssh_key=$(cat ${HOME}/.ssh/id_rsa.pub)
+    grep $ssh_key ~/.ssh/authorized_keys >/dev/null 2>&1
+    if [ $? -ne 0 ] ; then
+      if [ -f ~/.ssh/id_rsa.pub ] ; then
+        cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
+      else
+	echo "----------------------------------------------------------"
+	echo "WARNING: the deploy-ear.sh script will not work as you do not have a"
+	echo "public-private key"
```

Comment:
if the public/private key is used only to pull from the git repo, you could also use a git token and use the https repo link as such:
https://<token>@github.com/jimdowling/hopsworks-ee.git.
Git tokens would be, maybe, a bit more limiting maybe in what someone could do in case the data would be leaked.

Reply:
This is nothing to do with the Git repo. 
This is about deploying the hopsworks-ear.ear file from a dev machine (dev4.hops.works) to your vagrant machine.

Reply:
I tried deploying from my laptop to glassfish via a SSH bastion but the n/w bandwidth was ridiculously slow compared to going from the VM (vagrant) to localhost (dev4).